packages/libraries/core/src/client/http-client.ts

```diff
@@ -107,12 +109,16 @@ export async function makeFetchCall(
       const response = await (config.fetchImplementation ?? fetch)(endpoint, {
         method: config.method,
         body: config.body,
-        headers: config.headers,
+        headers: {
+          'x-request-id': requestId,
+          ...config.headers,
+        },
         signal,
       }).catch((error: unknown) => {
         const logErrorMessage = () =>
           logger?.error(
```

Comment:
In a future improvement, I think we should add the a request ID to the standard logger to avoid having to remember to include it in logs.
This would require we pass a context object into the logger or use another tool such as datadog to maintain a context for the request.